brion/spikeReport.cpp

```diff
@@ -147,12 +175,22 @@ std::future< Spikes > SpikeReport::read( float min )
             "Can't read to a time stamp inferior to the current time" ) );
     }
 
-    if ( _impl->threadPool.hasPendingJobs() )
+    if ( _impl->activeJobsCount )
     {
         LBTHROW( std::runtime_error( "Can't read : Pending read operation" ) );
     }
 
-    return _impl->threadPool.post( [&, min] { return _impl->plugin->read( min ); } );
+    auto spikes =
+        _impl->threadPool.post( [&, min] { return _impl->plugin->read( min ); } );
+    ++_impl->activeJobsCount;
+
+    auto result =
+        std::async( std::launch::deferred, commit< Spikes >, std::move( spikes ), [&] {
+            _impl->plugin->commitState();
+            --_impl->activeJobsCount;
+        } );
```

Comment:
Can you check if you can wait on this future with a timeout? I think that that doesn't work.
I don't think it's critical if it doesn't work, but it should be documented as an implementation limitation if that's the case.

After reading here: http://en.cppreference.com/w/cpp/thread/future/wait_for I think the answer is that wait_for will return deferred. In my opinion it's OK, but it makes wait_for kind of useless.